### PR TITLE
FINERACT-1724 - Replayed bulk transaction fix

### PR DESCRIPTION
**Pull request by b0c1**
FINERACT-1724 - Replayed bulk transaction fix
[x] - Remove bulk transaction from ReplayedTransactionBusinessEventService
[x] - Create Aspect and custom annotation for CommandHandler and Tasklet bulk events 
[x] - Add @BulkEventSupport annotation where ReplayedTransactionBusinessEventService was used 
[x] - Test class for Aspect call failure handling

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/aspect/BulkEventAspect.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Aspect
+public class BulkEventAspect {
+
+    private final BusinessEventNotifierService businessEventNotifierService;
+
+    @Around("execution(* org.apache.fineract.commands.handler.NewCommandSourceHandler.processCommand(..)) && @within(bulkEventSupport)")
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/aspect/BulkEventAspect.java]: I think we could remove the restriction to use this only on NewCommandSourceHandler interfaces, what if we just allow it everywhere?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/annotation/BulkEventSupport.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to be used to collect events into BulkEvent
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/annotation/BulkEventSupport.java]: Let's allow this on method level too.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyChargeToOverdueLoansBusinessStep.java [hunk 3]
```diff
@@ -28,6 +29,7 @@
 
 @Component
 @RequiredArgsConstructor
+@BulkEventSupport
```
Comment (ruchiD) [inline fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyChargeToOverdueLoansBusinessStep.java]: Can we verify that this setting is not conflicting global configuration setting for bulk events for Loan cob.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/aspect/BulkEventAspect.java (hunk 1 above)
Comment (b0c1) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/aspect/BulkEventAspect.java]: Modified using any class (with all methods) or only specific methods

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyChargeToOverdueLoansBusinessStep.java (hunk 3 above)
Comment (b0c1) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyChargeToOverdueLoansBusinessStep.java]: The aspect calls the already existing `BusinessEventNotifierService` methods that enable or disable the bulk logging flag. The flag is still checked in the `notifyPostBusinessEvent` method after the global flag check.

**Review by galovics — APPROVED** [review]

**Pull request merged**